admin-console/src/main/java/com/agiletec/apsadmin/portal/PageAction.java

```diff
@@ -637,14 +637,18 @@ protected String checkSetOffline(String selectedNode) throws ApsSystemException
 			return check;
 		}
 		IPage currentPage = this.getPage(selectedNode);
-		IPage[] childs = currentPage.getChildren();
+		String[] children = currentPage.getChildrenCodes();
 		IPage root = this.getPageManager().getOnlineRoot();
 		if (root.getCode().equals(currentPage.getCode())) {
 			this.addActionError(this.getText("error.page.offlineHome.notAllowed"));
 			return "pageTree";
-		} else if (null != childs && childs.length != 0) {
+		} else if (null != children && children.length != 0) {
 			boolean hasReferences = false;
-			for (IPage child : childs) {
+			boolean isOnline = currentPage.isOnline();
+			for (int i = 0; i < children.length; i++) {
+				IPage child = (isOnline)
+						? this.getPageManager().getOnlinePage(children[i])
+						: this.getPageManager().getDraftPage(children[i]);

```

Comment:
Maybe move this to a PageUtility object. Its pretty dense logic and it shows up several places in this class alone

Reply:
I do not understand what code should be moved; do you mean the code used to extract the page from the pageManager based on code and "isOnline" flag?

Reply:
Yeah. This block is repeated a lot.

    IPage child = (isOnline)
   ? this.getPageManager().getOnlinePage(children[i])
    : this.getPageManager().getDraftPage(children[i]);

It isn't critical but it shows up all over.